docs/content/Dialog2.mdx

```diff
@@ -105,8 +105,80 @@ The `DialogHeaderProps` interface extends `DialogProps` and adds these additiona
-| buttonType | `Button │ ButtonPrimary │ ButtonDanger` |         | The type of button to render                                                                                                                                       |
-| text       | `string`                                |         | Required. The button's inner text.                                                                                                                                 |
-| autoFocus  | `boolean`                               | false   | If true, this button will be automatically focused when the dialog is first rendered. If `autoFocus` is true on subsequent button definitions, it will be ignored. |
+| Prop name  | Type                                    | Default  | Description                                                                                                                                                        |
+| :--------- | :-------------------------------------- | :------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| buttonType | `Button │ ButtonPrimary │ ButtonDanger` | `Button` | The type of button to render                                                                                                                                       |
+| text       | `string`                                |          | Required. The button's inner text.                                                                                                                                 |
+| autoFocus  | `boolean`                               | false    | If true, this button will be automatically focused when the dialog is first rendered. If `autoFocus` is true on subsequent button definitions, it will be ignored. |
+
+## ConfirmationDialog
+
+A `ConfirmationDialog` is a special kind of dialog with more rigid behavior. It is used to confirm a user action. `ConfirmationDialog`s always have exactly two buttons: one to cancel the action and one to confirm it. No custom rendering capabilities are provided for ConfirmationDialog.
+
+### useConfirm hook
+
+An alternate way to use `ConfirmationDialog` is with the `useConfirm()` hook. It takes no parameters and returns an `async` function, `confirm`. When `confirm` is called (see ConfirmOptions below for its argument), it shows the confirmation dialog. When the dialog is dismissed, a promise is resolved with `true` or `false` depending on whether or not the confirm button was used.
+
+### Example (with `useConfirm` hook)
+
+```javascript live noinline
+function ShorthandExample2() {
+  const confirm = useConfirm()
+  const buttonClick = React.useCallback(
+    async function (e) {
+      // @todo The docs site live code examples don't work with async/await, so using the raw promise instead!
```

Comment:
It looks like you found a way to use async await instead of working with the raw promise, is this comment still valid?

Reply:
Whoops, forgot to remove the comment. Thanks!